services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java

```diff
@@ -376,9 +376,9 @@ public Response performLogin(@PathParam("provider_alias") String providerAlias,
             logger.debugf("Sending authentication request to identity provider [%s].", providerAlias);
         }
 
-        try {
-            AuthenticationSessionModel authSession = parseSessionCode(code, clientId, tabId, clientData);
+        AuthenticationSessionModel authSession = parseSessionCode(code, clientId, tabId, clientData);
```

Comment:
Issues in the tests are caused by this change. One option how to fix it, is to put the line back into the `try` block and the following `catch`:
```java
} catch (WebApplicationException e) {
    return e.getResponse();
}
```

Reply:
Interesting. How come in all the other endpoints that use `parseSessionCode(...)` the call isn't inside a try-catch? Maybe we have the same problem in the other endpoints, but it's not detected by tests?

Reply:
When this exception is thrown in an endpoint where it is not caught, it returns "400 Bad request" with some HTML. Maybe it's not ideal but don't see it as a big problem. Feel free to update just the original scope of this PR/issue you filled so the tests are green.

Reply:
Done and rebased on top of latest main.